nipype/interfaces/bids.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+""" Set of interfaces that allow interaction with BIDS data. Currently
+    available interfaces are:
+
+    BIDSDataGrabber: Query data from BIDS dataset using pybids grabbids.
+
+    Change directory to provide relative paths for doctests
+    >>> import os
+    >>> import bids
+    >>> filepath = os.path.realpath(os.path.dirname(bids.__file__))
+    >>> datadir = os.path.realpath(os.path.join(filepath, 'grabbids/tests/data/'))
+    >>> os.chdir(datadir)
+
+"""
+
+from .base import (traits,
+                   DynamicTraitedSpec,
+                   BaseInterface,
+                   isdefined,
+                   Str,
+                   Undefined)
+
+try:
+    from bids.grabbids import BIDSLayout
+except ImportError:
+    have_pybids = False
+else:
+    have_pybids = True
+
+from warnings import warn
+
+class BIDSDataGrabberInputSpec(DynamicTraitedSpec):
+    base_dir = traits.Directory(exists=True,
+                                desc='Path to BIDS Directory.',
+                                mandatory=True)
+    output_query = traits.Dict(key_trait=Str,
+                               value_trait=traits.Dict,
+                               desc='Queries for outfield outputs')
```

Comment:
I wonder if we should give this field a more sensible default - for example {"func": {"modality": "func"}, "anat": {"modality": "anat"}}. This would save time figuring out the syntax for 80% of users.

Reply:
I'm a fan. The current default of outputting all items is probably almost never usable.

Reply:
Here's an other idea for sensible defaults: If no infields are defined, infields for all BIDS entities in project are automatically created, and `Undefined` by default. If user connects an input to them, great, if not, they are not passed to pybids. It saves having to define infields.

Reply:
Sounds like a great idea!